src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -837,18 +835,21 @@ export class EditorComponent implements OnInit {
 
     const leftTopIndex = this.magicWandService.coordToDataArrayIndex(leftTop.x, leftTop.y, this.image.width);
     const rightBottomIndex = this.magicWandService.coordToDataArrayIndex(rightBottom.x, rightBottom.y, this.image.width);
-   
     let currRightTopIndex = this.magicWandService.coordToDataArrayIndex(rightBottom.x, leftTop.y, this.image.width);
+
     let newTopY = leftTop.y;
 
-    for (let i = leftTopIndex; i < paintedImageData.length; i += 4) {
+    for (let i = leftTopIndex; i <= rightBottomIndex; i += 4) {
       // If the alpha value has value.
-      if (paintedImageData[i + 3] === 255) {
+      if (paintedImageData[i + 3] > (255 / 2)) {
         paintedMask.add(i);
       }
 
-      // TODO(SHMCAFFREY) loop through only the indicies in the rectangle.
-
+      if (i === currRightTopIndex) {
+        // Minus 4 to account for i += 4 in loop.
+        i = this.magicWandService.coordToDataArrayIndex(leftTop.x, ++newTopY, this.image.width) - 4;
+        currRightTopIndex =  this.magicWandService.coordToDataArrayIndex(rightBottom.x, newTopY, this.image.width);
+      }
```

Comment:
This loop would probably be better written to loop over the (x,y) coordinates of the image and then convert to the indices which would avoid this complication/error prone logic.

No need to do this in this PR.  Just maybe add a TODO.